pyro/contrib/autoguide/initialization.py

```diff
@@ -1,35 +1,89 @@
+r"""
+The pyro.contrib.autoguide module contains initialization functions for
+automatic guides.
+
+The standard interface for initialization is a function that inputs a Pyro
+trace ``site`` dict and returns an appropriately sized ``value`` to serve
+as an initial constrained value for a guide estimate.
+"""
 from __future__ import absolute_import, division, print_function
 
 import torch
 from torch.distributions import transform_to
 
+from pyro.poutine.messenger import Messenger
+from pyro.util import torch_isnan
+
 
-@torch.no_grad()
 def init_to_feasible(site):
-    value = site["value"].detach()
+    """
+    Initialize to an arbitrary feasible point, ignoring distribution
+    parameters.
+    """
+    value = site["fn"].sample().detach()
     t = transform_to(site["fn"].support)
     return t(torch.zeros_like(t.inv(value)))
```

Comment:
This seems like you are taking inverse transform of `0` instead of `arbitrary`. What do you think about initializing uniformly in (-2, 2) in unconstrained space (which is Stan approach)?

Reply:
I think Stan's approach will not work well in Pyro. This zero=arbitrary function is added for backwards compatibility only. After this PR, all autoguides will init to the prior median.

Reply:
From my experience (with MAP inference for a bunch of small models), using median still does not work well if priors are not good enough. To resolve that, I did evaluate model's log_probs for a bunch of random init values and select the init value with largest log_prob. If you find that `median` does not work then you can give that method a try. :)

Reply:
> evaluate model's log_probs for a bunch of random init values

Sounds reasonable. If I end up having trouble with autoguide init, maybe we can share some machinery with HMC.